packages/twenty-server/src/database/commands/upgrade-version/0-32/0-33/0-33-update-rich-text-search-vector-expression.ts

```diff
@@ -0,0 +1,109 @@
+import { InjectRepository } from '@nestjs/typeorm';
+
+import chalk from 'chalk';
+import { Command } from 'nest-commander';
+import { Repository } from 'typeorm';
+
+import {
+  ActiveWorkspacesCommandOptions,
+  ActiveWorkspacesCommandRunner,
+} from 'src/database/commands/active-workspaces.command';
+import { Workspace } from 'src/engine/core-modules/workspace/workspace.entity';
+import {
+  FieldMetadataEntity,
+  FieldMetadataType,
+} from 'src/engine/metadata-modules/field-metadata/field-metadata.entity';
+import { SearchService } from 'src/engine/metadata-modules/search/search.service';
+import { WorkspaceMigrationRunnerService } from 'src/engine/workspace-manager/workspace-migration-runner/workspace-migration-runner.service';
+import {
+  NOTE_STANDARD_FIELD_IDS,
+  TASK_STANDARD_FIELD_IDS,
+} from 'src/engine/workspace-manager/workspace-sync-metadata/constants/standard-field-ids';
+import { FieldTypeAndNameMetadata } from 'src/engine/workspace-manager/workspace-sync-metadata/utils/get-ts-vector-column-expression.util';
+import { SEARCH_FIELDS_FOR_NOTES } from 'src/modules/note/standard-objects/note.workspace-entity';
+import { SEARCH_FIELDS_FOR_TASK } from 'src/modules/task/standard-objects/task.workspace-entity';
+
+@Command({
+  name: 'fix-0.33:update-rich-text-expression',
+  description: 'Update rich text (note and task) search vector expressions',
+})
+export class UpdateRichTextSearchVectorCommand extends ActiveWorkspacesCommandRunner {
+  constructor(
+    @InjectRepository(Workspace, 'core')
+    protected readonly workspaceRepository: Repository<Workspace>,
+    @InjectRepository(FieldMetadataEntity, 'metadata')
+    private readonly fieldMetadataRepository: Repository<FieldMetadataEntity>,
+    private readonly searchService: SearchService,
+    private readonly workspaceMigrationRunnerService: WorkspaceMigrationRunnerService,
+  ) {
+    super(workspaceRepository);
+  }
+
+  async executeActiveWorkspacesCommand(
+    _passedParam: string[],
+    _options: ActiveWorkspacesCommandOptions,
+    workspaceIds: string[],
+  ): Promise<void> {
+    this.logger.log('Running command to fix migration');
+
+    for (const workspaceId of workspaceIds) {
+      this.logger.log(`Running command for workspace ${workspaceId}`);
+
+      try {
+        const searchVectorFields = await this.fieldMetadataRepository.findBy({
+          workspaceId: workspaceId,
+          type: FieldMetadataType.TS_VECTOR,
+        });
+
+        for (const searchVectorField of searchVectorFields) {
+          let fieldsUsedForSearch: FieldTypeAndNameMetadata[] = [];
+          let objectNameForLog = '';
+
+          switch (searchVectorField.standardId) {
```

Comment:
I did not treat custom objects here as it is currently not possible to choose a rich text field as a label identifier, which is the condition for a field to be used in the search vector generation of such object.